org.adempiere.base/src/org/compiere/model/MDocType.java

```diff
@@ -304,6 +304,17 @@ public String getPrintName (String AD_Language)
 		return get_Translation (COLUMNNAME_PrintName, AD_Language);
 	}	//	getPrintName
 	
+	@Override
+	protected boolean beforeSave (boolean newRecord) {
+		if(newRecord || is_ValueChanged(COLUMNNAME_IsAutoGenerateInout) || is_ValueChanged(COLUMNNAME_IsAutoGenerateInvoice)) {
+			if(!DOCSUBTYPESO_PrepayOrder.equals(getDocSubTypeSO())) {
+				setIsAutoGenerateInout(false);

```

Comment:
Even if not used yet, in order to be clear I think is better to set the invoice flag to true for POS and OnCredit; and the shipment flag true for POS, OnCredit and Warehouse.
It would be weird to have the flag set to N in database and still the code is generating invoices and shipments